### PR TITLE
exact-count: refactor users.js and dynamic enable/disable

### PR DESCRIPTION
**Pull request by Weredime**
dynamicEnable/disable for all three
users.js refactor (it was messy)

### CODE REVIEW

**Review by apple502j — APPROVED** [review]

File: addons/exact-count/studio.js [hunk 1]
```diff
@@ -22,7 +22,21 @@ export default async function ({ addon }) {
     "https://api.scratch.mit.edu/studios/" + /[0-9]+/.exec(location.pathname)[0] + "/projects/?limit=40&offset=";
   const initialDelta = 100;
   const countElement = await addon.tab.waitForElement(".studio-tab-nav > a:first-child .tab-count");
+  const originalText = countElement.innerText;
+  let counted = false;
   countProjects(apiUrlPrefix, 0, initialDelta, function (count) {
-    countElement.innerText = `(${count})`;
+    counted = count;
```
Comment (WorldLanguages) [inline addons/exact-count/studio.js:28]: Could this code be converted from callbacks to promises, since we're refactoring anyway?

Comment (Weredime) [reply addons/exact-count/studio.js]: Done

File: addons/exact-count/forum.js [hunk 2]
```diff
@@ -1,12 +1,32 @@
...
   while (true) {
-    let userbox = await addon.tab.waitForElement(".postleft dl", { markAsSeen: true });
+    const userbox = await addon.tab.waitForElement(".postleft dl", { markAsSeen: true });
     if (userbox.childNodes[6] && userbox.childNodes[6].nodeValue.includes("+")) {
       const postCountReal = Number(/(\d+)\+? posts?/.exec(userbox.childNodes[6].nodeValue)[1]);
       fetch("https://scratchdb.lefty.one/v3/forum/user/info/" + userbox.querySelector("a").innerText)
         .then((response) => response.json())
         .then((data) => {
           if (data.counts && data.counts.total && data.counts.total.count > postCountReal) {
+            cache[userbox.parentElement.parentElement.parentElement.parentElement.id] = {
```
Comment (WorldLanguages) [inline addons/exact-count/forum.js:25]: Could we avoid `parentElement.parentElement`?

`userbox.closest(".blockpost")`?

**Review by WorldLanguages — APPROVED** [review]
LGTM
We need to be careful, if any addon also adds their own text to these elements, dynamically disabling exact-count could possibly remove text added by other addons.

File: addons/exact-count/forum.js (hunk 2 above)
Comment (Weredime) [reply addons/exact-count/forum.js]: Addressed

**Pull request merged**